include/envoy/init/init.h

```diff
@@ -34,6 +34,13 @@ class Manager {
    * each target at some point in the future.
    */
   virtual void registerTarget(Target& target) PURE;
+
+  enum class State { NotInitialized, Initializing, Initialized };
```

Comment:
nit: can we add doc comments for these? Sorry they were missing before (my bad) but worth adding now as this is a public interface.